src/hooks/useVotesRevealedEventsSummary.ts

```diff
@@ -0,0 +1,33 @@
+import { useContext } from "react";
+import { useQuery } from "react-query";
+import {
+  queryVotesRevealedEvents,
+  VoteRevealed,
+} from "web3/get/queryVotesRevealedEvents";
+import { ErrorContext } from "common/context/ErrorContext";
+import provider from "common/utils/web3/createProvider";
+import createVoidSignerVotingContractInstance from "web3/createVoidSignerVotingContractInstance";
+import determineBlockchainNetwork from "web3/helpers/determineBlockchainNetwork";
+
+const contract = createVoidSignerVotingContractInstance(
```

Comment:
can u explain how this works and why not live in a hook? I guess im worried that there's like some uncontrolled re-instantiation of the web3 provider happening, when ideally the entire app should use the same instance.

Reply:
Basically, we create a read only version of the contract to read data from the contract. We want this because the user can see data from the contract not logged in.